backend/pkg/report_emailer.go

```diff
@@ -22,72 +24,118 @@ func NewReportEmailer(sql *dbstore.SQLiteDatasource) *ReportEmailer {
 	return &ReportEmailer{sql: sql, inProgress: false}
 }
 
-func (re *ReportEmailer) configs() (*auth.AuthConfig, *auth.EmailConfig, int) {
-	authConfig := auth.NewAuthConfig(re.sql)
-	emailConfig := auth.NewEmailConfig(re.sql)
-	settings := re.sql.GetSettings()
+func (re *ReportEmailer) configs() (*auth.AuthConfig, *auth.EmailConfig, int, error) {
+	authConfig, err := auth.NewAuthConfig(re.sql)
+	if err != nil {
+		log.DefaultLogger.Error("ReportEmailer.configs: NewAuthConfig: " + err.Error())
+		return nil, nil, 0, err
+	}
+
+	emailConfig, err := auth.NewEmailConfig(re.sql)
+	if err != nil {
+		log.DefaultLogger.Error("ReportEmailer.configs: NewEmailConfig: " + err.Error())
+		return nil, nil, 0, err
+	}
 
-	return authConfig, emailConfig, settings.DatasourceID
+	settings, err := re.sql.GetSettings()
+	if err != nil {
+		log.DefaultLogger.Error("ReportEmailer.configs: GetSettings: " + err.Error())
+		return nil, nil, 0, err
+	}
+
+	return authConfig, emailConfig, settings.DatasourceID, nil
 }
 
 func (re *ReportEmailer) cleanup(schedules []dbstore.Schedule) {
-	// for _, schedule := range schedules {
-	// 	path := filepath.Join("data", schedule.ID+".xlsx")
-	// 	os.Remove(path)
+	log.DefaultLogger.Info("Starting Clean up...")
+	for _, schedule := range schedules {
 
-	// 	schedule.NextReportTime = int(time.Now().Unix()) + schedule.Interval
-	// 	re.sql.UpdateSchedule(schedule.ID, schedule)
-	// }
+		fileName := schedule.ID + ".xlsx"
+		log.DefaultLogger.Info(fmt.Sprintf("Deleting %s...", fileName))
+		path := filepath.Join("data", fileName)
+		err := os.Remove(path)
+		if err != nil {
+			// Shouldn't be toooo much of a problem since we're using the schedule ID for the report name, at the moment
```

Comment:
Just re. the comment, only thing I can think of is that maybe could possibly have permissions issues when run on Windows?